slowfast/models/optimizer.py

```diff
@@ -55,18 +55,18 @@ def construct_optimizer(model, cfg):
     optim_params = [
         {
             "params": bn_parameters,
-            "weight_decay": cfg.BN.WEIGHT_DECAY,
+            "weight_decay": cfg.SOLVER.WEIGHT_DECAY if cfg.FSDP.ENABLED else cfg.BN.WEIGHT_DECAY,
```

Comment:
This behavior doesn't seem good - we should always use the same config param to specify a setting. Is there a reason for doing this?

Reply:
This behaviour is to guarantee that all the optimizer's parameters should be the same within each FSDP wrapped layer with flattened parameters. Otherwise, this causes issues during training. I was wondering how this was handled in pycls btw?

Reply:
This is resolved for now. I'll submit another PR later to handle params groups + FSDP  and check for incompatible config.